### PR TITLE
Feature :: Integration rethinkDB

### LINKED ISSUE #2723: Integration :: RethinkDB 🚀
Is your proposal related to a problem?
Currently ToolJet doesn't have RethinkDB integration.

Describe the solution you'd like
Integration for RethinkDB.

**Supported Operations**

- [x] Delete database
- [x] Delete Table 
- [x] Create database
- [x] Create Table
- [x] List table
- [x] List database

- [x] Get  all documents
- [x] Insert table data 
- [x] Update  all table data
- [x] Update by id
- [x] Delete table data by id
- [x] Delete  all table data
- [x] Get document from primary key

**Authentication options**

Use the username and password to authenticate to the RethinkDB.

**Fields**

Host
Port
Protocol
Password
Username

If the feature is approved, would you be willing to submit a PR?
Yes .

### PR DESCRIPTION
**Pull request by stepinfwd**
closes #2723

### CODE REVIEW

File: plugins/packages/rethinkdb/lib/index.ts [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+          break;
+        }
+        case 'retreive_docs': {
+          result = await this.getDocumentByID(name, tablename, key, connection, database);
+          break;
+        }
+
+        case 'update_docs': {
+          result = await this.listAllTables(name, connection, database);
+          break;
```
Comment (gondar00) [inline plugins/packages/rethinkdb/lib/index.ts:60]: This case is conflicting with `list_table` case

Comment (stepinfwd) [reply plugins/packages/rethinkdb/lib/index.ts]: updated @gondar00

File: package.json [hunk 2]
```diff
@@ -31,6 +31,7 @@
   },
   "dependencies": {
     "@nestjs/cli": "^8.1.0",
+    "rethinkdb": "^2.4.2",
```
Comment (gsmithun4) [inline package.json:34]: `rethinkdb` should be added only to its corresponding plugin package

File: package-lock.json [hunk 3]
```diff
@@ -10149,6 +10166,21 @@
         "signal-exit": "^3.0.2"
       }
     },
+    "rethinkdb": {
```
Comment (gsmithun4) [inline package-lock.json:10169]: Same here

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

**Review by gsmithun4 — APPROVED** [review]
LGTM

**Pull request merged**